src/main/java/teammates/ui/webapi/action/DeleteFeedbackResponseCommentAction.java

```diff
@@ -0,0 +1,97 @@
+package teammates.ui.webapi.action;
+
+import teammates.common.datatransfer.attributes.FeedbackResponseAttributes;
+import teammates.common.datatransfer.attributes.FeedbackResponseCommentAttributes;
+import teammates.common.datatransfer.attributes.FeedbackSessionAttributes;
+import teammates.common.datatransfer.attributes.InstructorAttributes;
+import teammates.common.datatransfer.attributes.StudentAttributes;
+import teammates.common.exception.UnauthorizedAccessException;
+import teammates.common.util.Assumption;
+import teammates.common.util.Const;
+
+/**
+ * Deletes a feedback response comment.
+ */
+public class DeleteFeedbackResponseCommentAction extends Action {
+
+    @Override
+    protected AuthType getMinAuthLevel() {
+        return AuthType.LOGGED_IN;
+    }
+
+    @Override
+    public void checkSpecificAccessControl() {
+        long feedbackResponseCommentId = getLongRequestParamValue(Const.ParamsNames.FEEDBACK_RESPONSE_COMMENT_ID);
+        FeedbackResponseCommentAttributes frc =
+                logic.getFeedbackResponseComment(feedbackResponseCommentId);
+        if (frc == null) {
+            return;
+        }
+
+        String courseId = frc.courseId;
+
+        InstructorAttributes instructor = logic.getInstructorForGoogleId(courseId, userInfo.id);
+
+        if (instructor != null && frc.commentGiver.equals(instructor.email)) { // giver, allowed by default
+            return;
+        }
+
+        // Case 1: the delete request comes from instructor
+
+        if (instructor != null) {
```

Comment:
We might need something like `intent`  or just use `frc.isCommentFromFeedbackParticipant` to distinguish request from instructor or student? A person can be both instructor and student (this is true for demo data). In this case, he will never be able to delete his comment for responses he has given.

Reply:
> he will never be able to delete his comment for responses he has given

How is this not covered in the access control check before this? (giver, allowed by default)